### PR TITLE
signAndSubmitTransaction to accept InputTransactionData type to suppo…

### PR DESCRIPTION
**Pull request by 0xmaayan**
…rt wallet simulation

### CODE REVIEW

**Review by hardsetting — APPROVED** [review]

File: src/features/aptosSignAndSubmitTransaction.ts [hunk 1]
```diff
@@ -24,6 +25,12 @@ export type AptosSignAndSubmitTransactionMethod = (
...
-export type AptosSignAndSubmitTransactionInput = AnyRawTransaction
-/** Output of signing transactions. */
-export type AptosSignAndSubmitTransactionOutput = PendingTransactionResponse
+export interface AptosSignAndSubmitTransactionInput {
+  gasUnitPrice?: number;  // defaults to estimated gas unit price
+  maxGasAmount?: number;  // defaults to estimated max gas amount
+  payload: TransactionPayload | InputGenerateTransactionPayloadData;
+}
+
+export interface AptosSignAndSubmitTransactionOutput {
```
Comment (0xmaayan) [inline src/features/aptosSignAndSubmitTransaction.ts:34]: interesting, why this change? we might dont want to enforce the dapp to have the wallet waits for the transaction to be submitted?

File: src/features/aptosSignAndSubmitTransaction.ts [hunk 2]
```diff
@@ -24,6 +25,12 @@ export type AptosSignAndSubmitTransactionMethod = (
   transaction: AptosSignAndSubmitTransactionInput
 ) => Promise<UserResponse<AptosSignAndSubmitTransactionOutput>>
 
-export type AptosSignAndSubmitTransactionInput = AnyRawTransaction
-/** Output of signing transactions. */
-export type AptosSignAndSubmitTransactionOutput = PendingTransactionResponse
+export interface AptosSignAndSubmitTransactionInput {
+  gasUnitPrice?: number;  // defaults to estimated gas unit price
+  maxGasAmount?: number;  // defaults to estimated max gas amount
+  payload: TransactionPayload | InputGenerateTransactionPayloadData;
```
Comment (0xmaayan) [inline src/features/aptosSignAndSubmitTransaction.ts:31]: what is the reason for the different types...?

File: src/features/aptosSignAndSubmitTransaction.ts (hunk 1 above)
Comment (hardsetting) [reply src/features/aptosSignAndSubmitTransaction.ts:34]: We don't really need the whole `PendingTransactionResponse` object back typically.
The hash is all the dapp needs to fetch the status and was what we returned previously.
It's always easier to add more parameters later if they're required.

Comment (hardsetting) [reply src/features/aptosSignAndSubmitTransaction.ts:34]: I wouldn't be opposed to keeping `PendingTransactionResponse`, but it can be annoying i.e. for Petra we would have to change the code so that the additional fields are sent together with the `hash`

File: src/features/aptosSignAndSubmitTransaction.ts (hunk 2 above)
Comment (hardsetting) [reply src/features/aptosSignAndSubmitTransaction.ts:31]: `TransactionPayload` is the BCS-serializable class instance (you can build a `RawTransaction` with it)
`InputGenerateTransactionPayloadData` is the pojo format (cannot directly build a `RawTransaction` and have to fetch the ABI and convert first)

Comment (0xmaayan) [reply src/features/aptosSignAndSubmitTransaction.ts:31]: gotit. what are the different use cases that we would need both on the standard? 

If looking at the dapp<>adapter<>wallet flow, the dapp sends the json object -> adapter builds the transaction payload (either TransactionPayload or InputGenerateTransactionPayloadData )?

Comment (hardsetting) [reply src/features/aptosSignAndSubmitTransaction.ts:31]: the dapp might want to send the payload instance too

Comment (hardsetting) [reply src/features/aptosSignAndSubmitTransaction.ts:31]: we can start with just the pojo payload if you prefer, works for me

Comment (0xmaayan) [reply src/features/aptosSignAndSubmitTransaction.ts:31]: yes lets  do that, the adapter doesnt support the TransactionPayload type anyway
